org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/AbstractServiceReferenceAssert.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) OSGi Alliance (2019, 2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.assertj.servicereference;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+import org.assertj.core.api.AbstractAssert;
+import org.osgi.framework.ServiceReference;
+import org.osgi.test.assertj.dictionary.DictionaryAssert;
+
+public abstract class AbstractServiceReferenceAssert<SELF extends AbstractServiceReferenceAssert<SELF, ACTUAL>, ACTUAL extends ServiceReference<?>>
+	extends AbstractAssert<SELF, ACTUAL> {
+
+	protected AbstractServiceReferenceAssert(ACTUAL actual, Class<?> selfType) {
+		super(actual, selfType);
+	}
+
+	public DictionaryAssert<String,Object> serviceProperties() {
+		Dictionary<String, Object> props = new Hashtable<>();
+		if (actual.getPropertyKeys() != null) {
+			for (String key : actual.getPropertyKeys()) {
+				props.put(key, actual.getProperty(key));
+			}
+		}
+		return new DictionaryAssert<String, Object>(props);
+	}
+
```

Comment:
This is a good start, but it could do with a few more methods. A full-featured implementation would need:

* `isRegisteredInBundle()` (for `getBundle()`). Perhaps an overloaded version that takes a String so you can specify the BSN:version as a string rather than needing a `Bundle` instance to compare to.
* `isAssignableTo(Bundle, String)` (for the method of the same name). I would perhaps offer an overloaded version `isAssignableTo(Class<?>)`, and then use `FrameworkUtil.getBundle(Class)` and `Class.getName()` to pass on to `ServiceReference.isAssignableTo()`.
* `isBeingUsedBy(Bundle, Bundle...)` and `isBeingUsedBy(String, String...)` for `getUsingBundles()`

These don't have to be part of this PR as it is a bit out-of-scope for this PR. But at a minimum, please create a new issue to make sure we don't forget to finish fleshing out `ServiceReferenceAssert`.

Reply:
I added it to this PR